tests/Component.Tests/EventStreamTests.cs

```diff
@@ -0,0 +1,57 @@
+﻿using Component.Tests.Fakes;
+using Components;
+using FluentAssertions;
+using Microsoft.Framework.DependencyInjection;
+using System;
+using Xunit;
+
+namespace Component.Tests
+{
+    public class EventStreamTests
+    {
+        static IServiceProvider CreateProvider() => new ServiceCollection().AddInMemoryEventStream().BuildServiceProvider();
+        static EventStream CreateEventStream(IServiceProvider serviceProvider = null) => (serviceProvider ?? CreateProvider()).GetRequiredService<EventStream>();
+
+        [Fact]
+        public static void GivenANullObserverBasedSubscriberThrowArgumentNullException()
+        {
+            var stream = CreateEventStream();
```

Comment:
Should the `CreateEventStream` be tested in isolation then? `CreatedEventStream().Should().NotBeNull()`?